siriuspy/siriuspy/devices/bpm_eq.py

```diff
@@ -0,0 +1,581 @@
+        shape = (nbpm, 4, 2)
+        if not isinstance(gains, _np.ndarray):
+            gains = _np.full(shape, gains)
+        if gains.shape != shape:
+            raise ValueError(f'Wrong shape for gains. Must be {shape}')
+
+        for i, bpm in enumerate(self.bpms):
+            gns = gains[i]
+            for j, ant in enumerate('abcd'):
+                g = gns[j]
+                setattr(bpm, f'gain_direct_{ant}', g[0])
+                setattr(bpm, f'gain_inverse_{ant}', g[1])
+
+    @staticmethod
+    def round_gains(gains):
+        """Make gains compatible with 24 bits fixed point representation."""
+        mult = 1 << 24
+        return _np.int_(gains*mult) / mult
+
+    def acquire_bpm_data(self):
+        """."""
+        self._log('Starting Acquisition.')
+        self.data = dict()
+        self.data['timestamp'] = _time.time()
+        self.data['stored_current'] = self.currinfo.current
+        self.data['gains_init'] = self.get_current_gains()
+
+        init_source = self.trigger.source
+        self.trigger.source = self.trigger.source_options.index('Clock3')
+        try:
+            self._do_acquire()
+        except Exception as err:
+            self._log('ERR:Problem with acquisition:')
+            self._log(f'ERR:{str(err)}')
+
+        self.trigger.source = init_source
+        self.set_gains(self.data['gains_init'])
+        self._log('Acquisition Finished!')
+
+    def _do_acquire(self):
+        if self._acq_strategy == self.AcqStrategies.AssumeOrder:
+            self.set_gains(self.MAX_MULTIPLIER)
+        else:
+            gains = _np.full((len(self.bpms), 4, 2), self.MAX_MULTIPLIER)
+            gains[:, :, 1] *= self._acq_inverse_reduced_gain
+            self.set_gains(gains)
+        self.data['acq_strategy'] = self._acq_strategy
+        self.data['acq_inverse_reduced_gain'] = self._acq_inverse_reduced_gain
+
+        # acquire antennas data in FOFB rate
+        self._log('Preparing BPMs')
+        ret = self.cmd_mturn_acq_abort()
+        if ret > 0:
+            self._log(f'ERR: BPM {ret-1} did not abort previous acquistion.')
+            return
+
+        self.mturn_reset_flags_and_update_initial_timestamps()
+        ret = self.mturn_config_acquisition(
+            nr_points_after=self._acq_nrpoints, nr_points_before=0,
+            acq_rate='FOFB', repeat=False, external=True)
+        if ret > 0:
+            self._log(f'ERR: BPM {ret-1} did not start acquistion.')
+            return
+
+        self._log('Waiting BPMs to update')
+        ret = self.mturn_wait_update(timeout=self._acq_timeout)
+        if ret > 0:
+            self._log(f'ERR: BPM {ret-1} did not update in time.')
+            return
+        elif ret < 0:
+            self._log(f'ERR: Problem with acquisition. Error code {ret}')
+            return
+        self._log('BPMs updated.')
+
+        self._log('Acquiring data.')
+        fsamp = self.get_switching_frequency(1)
+        fswtc = self.get_sampling_frequency(1)
```

Comment:
`fsamp` and `fswtc` are switched (haha) here.

Reply:
thanks! Done!